### ISSUE TITLE
LaTeX-mode does not automatically initialize when opening a .tex file

### ISSUE THREAD

**Issue by variemai** [doomemacs/doomemacs#8191]
### I confirm that...

- [x] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [x] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [x] The issue can be reproduced on the **latest** available commit of Doom.

- [x] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

Opening a .tex file should initialize the AUCTEX major mode LaTeX-mode.  

### Current behavior

Opening a .tex file DOES NOT initialize the LaTeX-mode instead it defaults to the latex-mode. 

This behavior can be fixed by explicitly setting in the doom config file:

`(add-to-list 'auto-mode-alist '("\\.tex\\'" . LaTeX-mode))`

However, it used to work without explicitly setting it.

### Steps to reproduce

1. Run Emacs
2. Open a .tex file

### System Information

https://pastebin.com/98CdELJv

**Comment by mwoitek** [comment]
I agree that this behavior is a little inconvenient. But this isn't a bug. I had this problem when working on a multi-file project. Some of my files contained mostly text and almost no LaTeX. In those cases, LaTeX-mode wasn't enabled automatically. So I looked into this.

It turns out that, by default, AUCTeX inspects your TeX files to decide which major mode should be enabled. For example, if your file contains `\documentclass` then LaTeX-mode is selected. If you don't want the mode to be inferred, you can set a default and then force AUCTeX to use it:
```
(setq TeX-default-mode #'LaTeX-mode ;; should already be the default mode
      TeX-force-default-mode t)
```

**Comment by lizhuohua** [comment]
I recently upgraded Doom emacs and encountered the same issue.

I tried the suggested two workarounds, but neither worked, and I am sure that my tex file contains `\documentclass`.

**Comment by mwoitek** [comment]
@lizhuohua It's hard to tell what's happening without examining your config carefully. Try adding the following comment to the very top of your files: `% -*- mode: LaTeX; -*-`. This should not fail. If it does, then the problem probably originates in your config. Anyway, I don't believe this is Doom's fault.

**Comment by lizhuohua** [comment]
@mwoitek Thank you. After some investigation, I found that this bug was introduced in commit da08aa2d7d9ca9197896e7988dc8785006796336 and should only occur in Emacs version < 30.

This commit defines a new variable `major-mode-remap-defaults` introduced in Emacs 30, which makes AUCTeX mistakenly assume the current Emacs version is 30, and uses `major-mode-remap-defaults` for major mode redirection:

https://github.com/emacs-straight/auctex/blob/fc6ac87e5ed027e86b129a49d722897d127bbc50/tex-site.el#L127-L132

However, for Emacs version < 30, only `major-mode-remap-alist` is recognized, so my workaround is to add the following to my config:

```elisp
(setq major-mode-remap-alist major-mode-remap-defaults)
```

**Comment by mwoitek** [comment]
@lizhuohua Well, I stand corrected. In my defense, I haven't used LaTeX in a while.

**Comment by real-or-random** [comment]
Yep, the emulation introduced in da08aa2d7d9ca9197896e7988dc8785006796336 is incomplete, it should also add advice like the emulation for major-mode-remap-alist.

---

By the way, emacs 30 initializes `major-mode-remap-defaults` with mapping from the lowercase to uppercase modes: https://github.com/emacs-mirror/emacs/blob/68bcabf58afddb377d172857ce7648ad3ff09618/lisp/textmodes/tex-mode.el#L1041-L1045

This means that after loading AUXTeX, we have this:
```
((doctex-mode . docTeX-mode)
 (latex-mode . LaTeX-mode)
 (texinfo-mode . Texinfo-mode)
 (plain-tex-mode . plain-TeX-mode)
 (tex-mode . TeX-tex-mode)
 (LaTeX-mode . latex-mode)
 (plain-TeX-mode . plain-tex-mode)
 (TeX-mode . tex-mode))
```

Now this is full chaos. If you type `latex-mode` in `auto-mode-alist`, you'll get `LaTeX-mode`, and the other way around. I think AUCTeX should just remove the built-in remappings. (Note that even AUCTeX adds `LaTeX-mode` entries to `auto-mode-alist`.

I've just reported this to auctex-devel, let's see what they say.

**Comment by dr-scsi** [comment]
@real-or-random - FWIW, the issue you mentioned is fixed in AUCTeX sources with [this commit](https://git.savannah.gnu.org/cgit/auctex.git/commit/tex-site.el?id=717d9f733ea793ba40f3ee9e82a0f98aea9e0fbb).  Updating AUCTeX to version 14.0.9 which is available from ELPA should give you good results now (at least at AUCTeX's end, I can't tell about Doom Emacs).

**Comment by hlissner** [comment]
> FWIW, the issue you mentioned is fixed in AUCTeX sources with [this commit](https://git.savannah.gnu.org/cgit/auctex.git/commit/tex-site.el?id=717d9f733ea793ba40f3ee9e82a0f98aea9e0fbb)

Since `auctex` was bumped in 1bb191e8ebaf beyond this commit, I think we can assume this has been resolved in Doom as well. Thanks for letting us know!

**Comment by julienlabbe** [comment]
There are two issues in this thread.

The initial issue is not fixed with emacs versions prior to 30, but can be with
```
(setq major-mode-remap-alist major-mode-remap-defaults)
 ```
as shown by @lizhuohua in [its comment](https://github.com/doomemacs/doomemacs/issues/8191#issuecomment-2522039422).

The second issue is the mapping in `major-mode-remap-defaults` done by emacs from version 30, as pointed by @real-or-random [here](https://github.com/doomemacs/doomemacs/issues/8191#issuecomment-2588357022). This is the one that should be fixed after `auctex` update.

As it, the issue should not be closed.
